src/Sinch/Conversation/Messages/Message/ListMessage.cs

```diff
@@ -95,6 +97,31 @@ public interface IListItem
     {
     }
 
+    public class ListItem
```

Comment:
Not related to this dedicated PR, but there is a design point here (existing code is already having this e. g. https://github.com/sinch/sinch-sdk-dotnet/blob/main/src/Sinch/Conversation/Messages/Message/AppMessage.cs#L14)
This is the direct translation/representation of the oneOf from OAS
From a SDK user point of view, it is possible to fill both fields and send them to server, right ?
We should avoid this at SDK level and provide a way to fill a single field with the dedicated type
This subject is planning for next meeting where we could discuss more about it

Reply:
No, you can only send `ListItemChoise` or `ListItemProduct`, you cannot send both simultaneously.

This is achieved with private set and constructor which takes the specific type as a param. 

public default constructor is there, BUT `Obsolete` attribute with `error: True` use will result in a compiler error.

Reply:
Ok. Got it
Thank for C# growth/knowledge 👌